common/data_refinery_common/migrations/0038_paginate_default_svd_algorithm.py

```diff
@@ -1,26 +1,36 @@
 from django.db import migrations, models
-from django.core.paginator import Paginator
+
+def batch_update(queryset, batch=1000, **changes):
+    """ Update per batch """
+    is_done = False
+    current = 1
+
+    while not is_done:
+        start = (current - 1) * batch
+        end = start + batch
+        pks = queryset.values_list('pk', flat=True)[start:end][::1]
+        queryset.model.objects.filter(pk__in=pks).update(**changes)
+        is_done = len(pks) <= batch
+        if not is_done:
```

Comment:
I don't think this will terminate at the correct time. You're selecting `batch` ids each time, and then checking to see if they are less than or equal to `batch`. I would expect that condition to be true every loop. I think a better way might be to change the while condition to:
```
num_objects = queryset.count()
while end < num_objects:
```

Reply:
batch is a full page, so it wont be called again once fewer than 1000 is returned

Reply:
however i should update it to not call the update method on no objects.... since that is a waste

Reply:
What if exactly 1000 is returned? Which I think will be every loop.

Basically I think you have an off by one error but I suggested an implementation that avoids the possibility of an off by one error instead of just fixing it. But either solution works for me. But right now I'm pretty sure this only loops once because `len(pks) == batch` will be true the first time.

Reply:
oh it should be less than! good catch.